### PR TITLE
Update buildifier/buildozer instructions

### PR DESCRIPTION
**Pull request by tpetkov-VMW**
As of Go 1.17 , installing packages with go get is deprecated
See https://go.dev/doc/go-get-install-deprecation
Signed-off-by: Toma Petkov <tpetkov@vmware.com>

Commit Message: bazel: Update buildifier/buildozer instructions
Additional Description: As of Go 1.17 , installing packages with go get is deprecated. See https://go.dev/doc/go-get-install-deprecation
Risk Level: Low
Testing:
```sh
go install github.com/bazelbuild/buildtools/buildifier@latest
go install github.com/bazelbuild/buildtools/buildozer@latest
```
Docs Changes: Changed instructions for buildifier/buildozer
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by tpetkov-VMW** [comment]
Not sure if I should update tools/code_format/paths.py to check inside  `$HOME/go/bin/` directory.
Reviewers - is there a reason not to?

**Comment by phlax** [comment]
relatedly i just removed the buildozer/buildifier rules for bazel (on the basis of not currently being used) - which would i think be a preferable way to do it

not sure re code_format but i can take a look later today

**Comment by phlax** [comment]
not sure if there is something else required (im not a go expert - so maybe this is something obvious)

i tried the following steps in a `debian:slim-bullseye` container

```console
$ apt update
$ apt install golang
$ go install github.com/bazelbuild/buildtools/buildifier@latest
```
which gave me the following error:

```console
package github.com/bazelbuild/buildtools/buildifier@latest: cannot use path@version syntax in GOPATH mode

```

```

**Comment by tpetkov-VMW** [comment]
For installation, I used the instructions here: https://go.dev/doc/install
```sh
go version
go version go1.18 linux/amd64
```
From reading a bit about it online, you'd have to use go install for version 1.17 onward.
@phlax - My guess is you have go1.16 , if that's the case, then maybe we should specify either:
An exact way to install golang, to enforce a single version.
Two diverging installation routes, depending on the version.

**Comment by phlax** [comment]
im ok with just specifying the required go version (im guessing bullseye backports would have that if required) on the basis that i think in the medium term we move this to bazel anyway

**Comment by phlax** [comment]
~related this is the currently specified version used in CI (and by bazel)

```console
$ git grep GO_VERSION
bazel/dependency_imports.bzl:GO_VERSION = "1.17.5"
```

**Comment by tpetkov-VMW** [comment]
```sh
wget https://go.dev/dl/go1.17.5.linux-amd64.tar.gz
rm -rf /usr/local/go && tar -C /usr/local -xzf go1.17.5.linux-amd64.tar.gz
rm /root/go/bin/build*
go version
go version go1.17.5 linux/amd64
go install github.com/bazelbuild/buildtools/buildifier@latest
go install github.com/bazelbuild/buildtools/buildozer@latest
```
Tried version 1.17.5 and the commands do work.
I'll just specify in the docs, that at least 1.17.5 is needed

File: bazel/README.md [hunk 1]
```diff
@@ -260,6 +260,7 @@ for how to update or override dependencies.
 
 1. Install Golang on your machine. This is required as part of building [BoringSSL](https://boringssl.googlesource.com/boringssl/+/HEAD/BUILDING.md)
    and also for [Buildifer](https://github.com/bazelbuild/buildtools) which is used for formatting bazel BUILD files.
+   Make sure you have go version 1.17.5 or later.
```
Comment (phlax) [inline bazel/README.md:263]: im guessing we can just make this `1.17`

Comment (moderation) [reply bazel/README.md:263]: `1.17.x` ?

**Review by phlax — APPROVED** [review]
lgtm, thanks @tpetkov-VMW

**Pull request merged**